### ISSUE TITLE
mypy's --pretty option printing one word per line when used with pre-commit

### ISSUE THREAD

**Issue by deveshks** [python/mypy#8816]
On running mypy as a [pre-commit](https://github.com/pre-commit/pre-commit) hook along with `--pretty` option, mypy prints one word per line.

For example, after having the following `pre-commit` configuration with `pip` which includes the `--pretty` option

https://github.com/pypa/pip/blob/d911a9fcb8d29ed560ba0ccb16f83f914a37a2a7/.pre-commit-config.yaml#L31-L41

If I introduce a mypy error, it looks like:

```
$ pre-commit run mypy --files src/pip/_internal/cli/main_parser.py
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

src/pip/_internal/cli/main_parser.py:47: error:
"ConfigOptionParser"
has
no
attribute
"main"
        parser.main = True
        ^
Found 1 error in 1 file (checked 1 source file)

mypy, for Python 2.......................................................Failed
- hook id: mypy
- exit code: 1

src/pip/_internal/cli/main_parser.py:47: error:
"ConfigOptionParser"
has
no
attribute
"main"
        parser.main = True
        ^
Found 1 error in 1 file (checked 1 source file)
```

**Comment by pradyunsg** [comment]
This is eerily similar to https://github.com/python/mypy/issues/8144, which was filed as a result of a discussion between @asottile and myself, about a pre-commit + mypy's `--pretty` not working well together.

**Comment by asottile** [comment]
I believe that only got fixed in v0.770 and it looks like pip is still on v0.760

**Comment by deveshks** [comment]
> I believe that only got fixed in v0.770 and it looks like pip is still on v0.760

Yep, I updated https://github.com/pypa/pip/blob/d911a9fcb8d29ed560ba0ccb16f83f914a37a2a7/.pre-commit-config.yaml#L32 to ` v0.770` and see the following now

```
$ pre-commit run mypy --files src/pip/_internal/cli/main_parser.py
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

src/pip/_internal/cli/main_parser.py:47: error: "ConfigOptionParser" has no
attribute "main"
        parser.main = True
        ^
Found 1 error in 1 file (checked 1 source file)

mypy, for Python 2.......................................................Failed
- hook id: mypy
- exit code: 1

src/pip/_internal/cli/main_parser.py:47: error: "ConfigOptionParser" has no
attribute "main"
        parser.main = True
        ^
Found 1 error in 1 file (checked 1 source file)
```

**Comment by ethanhs** [comment]
Thanks @pradyunsg and @asottile!
